source/extensions/filters/http/grpc_json_transcoder/descriptor_pool_builder.cc

```diff
@@ -0,0 +1,156 @@
+#include "source/extensions/filters/http/grpc_json_transcoder/descriptor_pool_builder.h"
+
+using Envoy::Protobuf::FileDescriptorSet;
+using std::string;
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcJsonTranscoder {
+
+void DescriptorPoolBuilder::requestDescriptorPool(
+    std::function<void(std::unique_ptr<Protobuf::DescriptorPool>&&)> descriptor_pool_available) {
+  descriptor_pool_available_ = descriptor_pool_available;
+  FileDescriptorSet descriptor_set;
+
+  switch (proto_config_.descriptor_set_case()) {
+  case envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder::
+      DescriptorSetCase::kProtoDescriptor:
+    if (!descriptor_set.ParseFromString(
+            api_->fileSystem().fileReadToEnd(proto_config_.proto_descriptor()))) {
+      throw EnvoyException("transcoding_filter: Unable to parse proto descriptor");
+    }
+    loadFileDescriptorProtos(descriptor_set);
+    break;
+  case envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder::
+      DescriptorSetCase::kProtoDescriptorBin:
+    if (!descriptor_set.ParseFromString(proto_config_.proto_descriptor_bin())) {
+      throw EnvoyException("transcoding_filter: Unable to parse proto descriptor");
+    }
+    loadFileDescriptorProtos(descriptor_set);
+    break;
+  case envoy::extensions::filters::http::grpc_json_transcoder::v3::GrpcJsonTranscoder::
+      DescriptorSetCase::kReflectionClusterConfig:
+    // The constructor of this object is responsible for ensuring the async_reflection_fetcher_ is
+    // not null.
+    async_reflection_fetcher_->requestFileDescriptors(
```

Comment:
Ideally,  async_reflection_fetcher should be created here, as it is implementation detail of DescriptorPoolBuilder.   For easy unit-test to pass in a mock async_fetcher,  you can use a asyn_fetch_factory, so it can be mocked.

Reply:
number of lines doesn't matter here,  it is where the logic belongs matter.    A module design with clear logic in separate modules is much better than codes lump all logic into one function.

Reply:
For the record, I believe this is a matter of software design philosophy, but I always conform to local style, so I'll update this since it sounds like you feel strongly about it.